src/main/surfrdUtilsMod.F90

```diff
@@ -219,7 +302,7 @@ subroutine collapse_to_dominant(weight, lower_bound, upper_bound, begg, endg, n_
        end do
 
        ! Error checks
-       call check_sums_equal_1(weight, begg, 'weight', subname)
+       if (wt_sum == 1._r8) call check_sums_equal_1(weight, begg, 'weight', subname)
```

Comment:
Why is this only being done when wt_sum==1?

Reply:
@ekluzek , I'm adding this comment to the code:

Check that weight sums to 1 only if weight summed to 1 upon entering the subroutine. If weight did not sum to 1 upon entering the subroutine, then weight will not sum to 1 now, either. Currently the latter occurs when collapsing the urban landunits to the dominant urban landunit.

Reply:
@slevisconsulting let me make sure I understand here. If you are collapsing urban landunits, wt_sum won't be equal to 1. upon entry or exit to this subroutine. At what point does that get rectified? When the model starts up wt_sum is going to sum to 1.0 right?

Reply:
...sorry for the delay in my response:

@ekluzek , yes, if you are collapsing urban landunits, wt_sum will be less than or equal to 1, because the urban landunits are only three of the model's landunits.

The code encounters this statement a second time if n_dom_landunits > 0 and < 9. This time wt_sum = 1 because now it's working with all the landunits.

One way to remove the `if` in front of the error check might be to move the error check outside of the subroutine call. Do you prefer that?

Reply:
To be exact, I could see placing the error check without the `if` in subr. surfrd_get_data after each call:
- call collapse_to_dominant (if collapse_urban is true)
- call collapse_to_dominant (second call, this time for n_dom_landunits)
- call collapse_individual_lunits

Reply:
@ekluzek , I will go ahead and try this since I haven't heard from you. We can always change back if you don't like the change.

Reply:
Hey Sam. I looked into the code a bit more. I think maybe the thing to do would be to have a if else structure for the check_sums_equal_1 call. If the previous sum was 1, you call it normally, but otherwise you send it the previous sum as the optional argument "sumto". That way it's still always doing this check. In the end I think that's what we want, for the appropriate check to be done each time. So as you say another option is to pull the checkout out of the collapse subroutine and then you can make sure the right check is done afterwards. The bad thing there is that then if the collapse call is used somewhere else, you also need to bring the call to the check at the same time.

Reply:
@ekluzek , we have the following conundrum...

When `collapse_urban = .true.` the wt_sum has a different value < 1 for each g, because the sum of urban is different in every grid cell.

This means that I need to call the error check within a `do g` loop. However, now I get the error:
`The shape matching rules of actual arguments and dummy arguments have been violated`

I'm not sure if this could be fixed by changing this...
`real(r8)        , intent(in) :: arr(lb:,:)   ! array to check`
to this...
`real(r8)        , intent(in) :: arr(lb:ub,:)   ! array to check

I could try it, but this would require changing the call arguments to include `ub` in all existing calls to this subroutine. So I wanted to run it by you before trying.

A different alternative would be to change `sumto` from scalar to an array.

Reply:
It looks like sumto isn't used in any other calls (that I could see) of check_sums_equal_1, so I'd say change sumto to an array.

Reply:
I found it here only:
`main/surfrdUtilsMod.F90:          call check_sums_equal_1(wt_cft, begg, 'wt_cft', subname//': irrigation', sumto=TotalSum)`
`main/surfrdUtilsMod.F90:       call check_sums_equal_1(wt_cft, begg, 'wt_cft', subname//': mergetoclmpft', sumto=TotalSum)`

...and maybe this will still work even with sumto as an array?

Reply:
TotalSum would need to be an array as well. Which might not be hard to do..

Reply:
The unit tests for it would also need to be updated...

Reply:
Done. See latest push.